egs/wsj/s5/steps/nnet3/chain/train.py

```diff
@@ -343,7 +343,7 @@ def TrainNewModels(dir, iter, srand, num_jobs, num_archives_processed, num_archi
     if left_deriv_truncate is not None:
         deriv_time_opts += " --optimization.min-deriv-time={0}".format(left_deriv_truncate)
     if right_deriv_truncate is not None:
-        deriv_time_opts += " --optimization.max-deriv-time={0}".format(int(chunk-width-right_deriv_truncate))
+        deriv_time_opts += " --optimization.max-deriv-time={0}".format(int(chunk_width - 1 - right_deriv_truncate))
```

Comment:
I guess left-deriv-truncate was originally intended to be non-negative to truncate the deriv within chunk_width. Anyway, I have made changes to add --trainer.deriv-truncate-margin